src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -3498,16 +3498,11 @@ internal Collection<PSObject> NewItem(string[] paths, string name, string type,
 
                     if (isSymbolicJunctionOrHardLink)
                     {
-                        if (content == null)
-                        {
-                            throw PSTraceSource.NewArgumentNullException(SessionStateStrings.NewItemValueNotSpecified, path);
-                        }
-
-                        string targetPath = content.ToString();
+                        string targetPath;
 
-                        if (string.IsNullOrEmpty(targetPath))
+                        if (content == null || string.IsNullOrEmpty(targetPath = content.ToString()))
```

Comment:
Please use new operator:
```suggestion
                        if (content is null || string.IsNullOrEmpty(targetPath = content.ToString()))
```